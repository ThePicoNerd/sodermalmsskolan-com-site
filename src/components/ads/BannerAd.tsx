import React, { FunctionComponent } from "react";
import Ad, { AdProps } from "./Ad";

export type BannerAdProps = Partial<AdProps>;

/**
 * A banner ad.
 *
 * @param {React.PropsWithChildren<BannerAdProps>} props Props.
 *
 * @returns {React.ReactElement} A rendered banner ad.
 */
const BannerAd: FunctionComponent = (props) => (
  <Ad
    format="auto"
    fullWidthResponsive
    {...props}
  />
);

export default BannerAd;
