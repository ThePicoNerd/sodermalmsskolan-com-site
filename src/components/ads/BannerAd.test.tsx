import React from "react";
import { render } from "@testing-library/react";
import BannerAd from "./BannerAd";

test("<BannerAd />", () => {
  const result = render(<BannerAd />);

  expect(result.asFragment()).toMatchSnapshot();
});
