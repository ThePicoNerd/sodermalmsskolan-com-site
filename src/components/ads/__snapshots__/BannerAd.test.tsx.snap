// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerAd /> 1`] = `
<DocumentFragment>
  <ins
    class="block adsbygoogle"
    data-ad-client="ca-pub-1944185766034883"
    data-ad-format="auto"
    data-ad-slot="5909725353"
    data-full-width-responsive="true"
  />
</DocumentFragment>
`;
